src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -704,6 +704,15 @@ export const selectAccountIdsByAssetId = createCachedSelector(
     `${state.portfolio.connectedWallet?.id}-${filter?.assetId}` ?? 'assetId',
 )
 
+export const selectAccountIdsByChainId = createCachedSelector(
```

Comment:
question about the cached selector. if we are caching based on wallet id and chain id, it seems like we would get stale data when fetching account ids for the same wallet/chain after a runtime update like adding snaps or importing more accounts from ledger. is this accurate, or does changes to portfolio accounts auto invalidate the cached selector?

Reply:
Yeah, that's a direct copy and paste job from https://github.com/shapeshift/web/pull/3104.

In effect, we're actually safe however. See an explanation from LLM robots below after feeding them the README from reselect and our implementation. tl;dr the misconception (which I fall for all the time myself 🤣) is assuming the cache key is the *only* source of referential in/validation, where in reality it's cache key *+ input selectors*. And indeed, accounts being added or deleted for a given ChainId (e.g on snap un/installed) is happy at runtime:



https://github.com/shapeshift/web/assets/17035424/5082a255-5568-4577-8c1d-08533a8f30c8


## Caching Mechanism Explained

### How Caching Works

The `createCachedSelector` function enhances selector performance in Redux by memorizing and caching results based on specific arguments. This optimization prevents redundant calculations when input data remains unchanged, ensuring efficient data retrieval.

#### Your `selectAccountIdsByChainIdFilter` Selector

- **Takes a list of portfolio accounts** and a **filter object** from the Redux state.
- Applies the `findAccountsByChainId` function to these inputs to **filter accounts by a specified chain ID**.
- Utilizes a **cache key** derived from the connected wallet's ID and the chain ID from the filter.

### Cache Key and Its Role

- The cache key is essential for determining whether to return a cached result or recompute the selector.
- Constructed from a combination of the wallet ID and the chain ID, or defaults to 'chainId' if those aren't available.
- Ensures that each unique combination of wallet ID and chain ID has a separate cache entry.

### Addressing Potential Stale Data Concerns

Concerns regarding stale data may arise with runtime updates (e.g., adding snaps or importing more accounts) that alter the underlying accounts but not the wallet ID or chain ID.

#### Key Insight: Dependency Intelligence

- Selectors' effectiveness is directly tied to their dependencies.
- This selector depends on:
  - `selectPortfolioAccounts`: Returns the state's list of accounts.
  - `selectChainIdParamFromFilter`: Extracts the chain ID from a filter object.

- If any input changes (the list of accounts or the chain ID), `re-reselect` detects this change. Given that the cache key depends on these inputs, any significant alteration will invalidate the relevant cache entry and trigger a re-computation with the new data.

## Conclusion

Provided that all changes affecting the cache (like modifying account lists) are reflected in the selector's input selectors (`selectPortfolioAccounts`, `selectChainIdParamFromFilter`), the caching mechanism will function correctly. It won't return stale data because it will recompute the selector whenever its inputs change in a manner that affects the output. This automatic cache invalidation ensures that your selector always returns up-to-date information based on the latest state.

Reply:
Baller response and explanation robots! :beers: